pandas/core/apply.py

```diff
@@ -131,10 +131,11 @@ def get_result(self):
 
         # ufunc
         elif isinstance(self.f, np.ufunc):
-            with np.errstate(all='ignore'):
-                results = self.f(self.values)
-            return self.obj._constructor(data=results, index=self.index,
-                                         columns=self.columns, copy=False)
+            result = self.obj._constructor(index=self.index, copy=False)
+            for col in self.columns:
+                with np.errstate(all='ignore'):
```

Comment:
When given a list of series, the `DataFrame` constructor converts them all to arrays, so the columns can't all be passed into the constructor at once. To prevent inefficient constructing in the common non-sparse case, how about checking whether there are sparse columns at all, and if there are then the construction happens column-by-column but if there aren't then it does what it did before?

Reply:
We need to be careful here. Previously, for a homogenous DataFrame of non-extension array values, `df.apply(ufunc)` would result in one call to the ufunc.

If we go columnwise, we'll have `n` calls to the ufunc.

Should this be done block-wise and then the results stitched together?